external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisReceiver.scala

```diff
@@ -221,6 +221,12 @@ private[kinesis] class KinesisReceiver[T](
     }
   }
 
+  /** Return the current rate limit defined in [[BlockGenerator]]. */
+  private[kinesis] def getCurrentLimit: Int = {
+    assert(blockGenerator != null)
```

Comment:
I would be okay to keep it if we add a useful error in case this assertion doesn't hold, e.g.
`assert(blockGenerator != null, "Expected blockGenerator to be set for the receiver before the processor received records")`

or something like that